### DISCUSSION TITLE
example using falcon

### DISCUSSION THREAD

**Discussion by YuriFontella** [miguelgrinberg/python-socketio#1207] [General]
Hello, I already have an api with the falcon framework, now I'm trying to implement a socketio server. Can this be possible? 

If anyone can show me a way.

**Comment by miguelgrinberg** [comment]
Falcon is a WSGI framework, so you can use the WSGI middleware provided in this package to integrate it with a Socket.IO application. Quick example:

```python
import socketio
from api import app as falcon_app   # <-- this is your falcon app

sio = socketio.Server()
app = socketio.WSGIApp(sio, falcon_app)  # <-- this is the integrated WSGI app
```

With this, you now run the combined `app` instance instead of your original falcon app and that will route traffic to both your falcon app and the Socket.IO app depending on the URL.

**Comment by YuriFontella** [comment]
Thanks for answering.
The wsgi integration worked great and the requests to the routes also as expected.

The problem was when I connected a client, I may have done something wrong.

But this message appeared and I don't know if it's related to falcon or socketio. And the requests were extremely slow.
Can you see a problem here?

![image](https://github.com/miguelgrinberg/python-socketio/assets/4920619/ae438b67-c95f-4a84-a7ae-e36bc16633b7)

I'm starting with gunicorn

**Reply by miguelgrinberg** [reply]
Try installing the `simple-websocket` package. In addition, check the documentation to learn how to start Gunicorn in threaded mode, if you aren't doing that already.

**Reply by YuriFontella** [reply]
reading the documentation I made some changes and everything is working perfectly 

`socket = socketio.Server(async_mode='threading')` I don't know if it's really necessary
`gunicorn --threads 100 srv.www:app`

thanks for the attention and help @miguelgrinberg
